src/dotnet/UI.Blazor/Services/ModalUI.cs

```diff
@@ -48,7 +36,7 @@ public async Task<ModalRef> Show<TModel>(TModel model, ModalOptions options)
         ModalOptions options)
         where TModel : class
     {
-        await WhenReady;
+        await WhenReady.ConfigureAwait(false);
```

Comment:
The code below seem to require synchronization context.

Reply:
done